crates/net/rpc/src/engine/mod.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{
+    result::{rpc_err, ToRpcResult},
+    EthApiSpec,
+};
+use async_trait::async_trait;
+use jsonrpsee::core::RpcResult as Result;
+use reth_consensus::Config;
+use reth_interfaces::consensus::ForkchoiceState;
+use reth_primitives::{H256, H64};
+use reth_rpc_api::EngineApiServer;
+use reth_rpc_types::engine::{
+    ExecutionPayload, ForkchoiceUpdated, PayloadAttributes, PayloadStatus, PayloadStatusEnum,
+    TransitionConfiguration,
+};
+
+mod error;
+pub use error::EngineApiError;
+
+/// TODO:
+pub struct EngineApi {
+    /// The implementation of `eth` API
+    eth: Box<dyn EthApiSpec>,
```

Comment:
I think generic better? @mattsse

Reply:
probably doesn't really matter, but this would make it easier to reuse `EngineApi` as a standalone element, without having to deal with additional types.